APIMF-3048: add javadoc and remove private[amf]

arielmirra opened the pull request:
- Add javadoc to most remodularization classes, methods and objects
- Remove the private[amf] modifier on those classes/methods/objects

Commit:
APIMF-3048: add javadoc and remove private[amf]

- Add javadoc to most remodularization classes, methods and objects
- Remove the private[amf] modifier on those classes/methods/objects

tomsfernandez commented on amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala:
```diff
@@ -6,25 +6,44 @@ import amf.plugins.document.vocabularies.model.document.{Dialect, DialectInstanc
-private[amf] class AMLClient(protected override val configuration: AMLConfiguration)
+/**
+  * AML Client object
+  * @param configuration {@link amf.client.environment.AMLConfiguration}
```
Doesn't make sense if constructor is private ?

tomsfernandez replied:
The java doc param i mean

arielmirra replied:
maybe, but it enables navigation through the JavaDoc that was the main purpose.

tomsfernandez replied:
Hmm dunno it's weird. Javadocs are for client purposes and if the constructor is private (clients do not know how to build it) then why document it for clients?

If someone that wants to know how to build it, aka an AMF developer, wants to know how it is built then he will look at the code because he will be working with it. He will not treat is as a black-box as a client would.

My stance is that we should remove this. Suppose @Pope141 is the tie-breaker.

pope1838 replied:
I want to left aside the java dog for a moment. We want that AMLClient constructor be private, so we can handle the instance creation and be sure that it come always from a configuration. (At least that was the original idea).
So, the private constructor for me is ok.
Having that in mind, we can discuss about the java doc. Of course the user will not be able to create an instance but the purpose of the doc is to document for what the client is (regarding which information contains, as it cannot be builded externally). Maybe, we could just have a text describing the motivation of the client, basically a class to provide and concentrate common use cases and handled typed results (As is the case of parseDialect for instance)

arielmirra replied:
Taking your feedback into account I've:

- Left the private constructor
- Modified the JavaDoc to explain what the Client is and what it is used for

Commit:
APIMF-3048: apply feedback from PR

- Left the private constructor
- Modified the JavaDoc to explain what the Client is and what it is used for
